src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,26 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        model_name = None
+        if isinstance(self, sagemaker.model.FrameworkModel):
```

Comment:
what if it's an instance of model? what happens? We need to support that case too right?